presto-spi/src/main/java/com/facebook/presto/spi/ColumnMetadata.java

```diff
@@ -33,60 +33,29 @@ public class ColumnMetadata
     private final String name;
     private final Type type;
     private final boolean nullable;
-    private final String comment;
-    private final String extraInfo;
+    private final Optional<String> comment;
+    private final Optional<String> extraInfo;
     private final boolean hidden;
     private final Map<String, Object> properties;
 
-    public ColumnMetadata(String name, Type type)
+    private ColumnMetadata(Builder builder)
     {
-        this(name, type, true, null, null, false, emptyMap());
+        this(builder.name, builder.type, builder.nullable, builder.comment, builder.extraInfo, builder.hidden, builder.properties);
     }
 
-    /**
-     * @deprecated Use {@link #builder()} instead.
-     */
-    @Deprecated
-    public ColumnMetadata(String name, Type type, String comment, boolean hidden)
-    {
-        this(name, type, true, comment, null, hidden, emptyMap());
-    }
-
-    /**
-     * @deprecated Use {@link #builder()} instead.
-     */
-    @Deprecated
-    public ColumnMetadata(String name, Type type, String comment, String extraInfo, boolean hidden)
-    {
-        this(name, type, true, comment, extraInfo, hidden, emptyMap());
-    }
-
-    /**
-     * @deprecated Use {@link #builder()} instead.
-     */
-    @Deprecated
-    public ColumnMetadata(String name, Type type, String comment, String extraInfo, boolean hidden, Map<String, Object> properties)
-    {
-        this(name, type, true, comment, extraInfo, hidden, properties);
-    }
-
-    /**
-     * @deprecated Use {@link #builder()} instead.
-     */
-    @Deprecated
-    public ColumnMetadata(String name, Type type, boolean nullable, String comment, String extraInfo, boolean hidden, Map<String, Object> properties)
+    protected ColumnMetadata(String name, Type type, boolean nullable, Optional<String> comment, Optional<String> extraInfo, boolean hidden, Map<String, Object> properties)
```

Comment:
should this be private as well?

Reply:
PinotColumnMetadata is extending it, so I had made this protected - https://github.com/prestodb/presto/pull/24616/files#diff-a793fc0f428838efb2971ab04c72c665638c2f842458a45e3b741f86cadc4d52R33